### PR TITLE
PAY-3831 : helm chart upgrade changes and spring boot 2.3.0 upgrade

### PR DESCRIPTION
**Pull request by prashant-khunte**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -69,7 +71,8 @@ dependencies {
     compile group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: javaLoggingVersion
     compile group: 'io.springfox', name: 'springfox-swagger2', version: '2.9.0'
     compile group: 'io.springfox', name: 'springfox-swagger-ui', version: '2.9.0'
-    compile group: 'uk.gov.hmcts.reform', name: 'health-spring-boot-starter', version: '0.0.3'
+    // Removed health starter for spring boot 2.3.0 and helm chart upgrade PAY-3831
```
Comment (amitchidrewar1301) [inline api/build.gradle:74]: Please remove commented code

File: api/build.gradle [hunk 2]
```diff
@@ -58,18 +68,19 @@ dependencies {
     compile group: 'org.springframework.boot', name: 'spring-boot-starter-web'
     compile group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa'
     compile group: 'org.springframework.boot', name: 'spring-boot-starter-security'
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-actuator'
+    compile (group: 'org.springframework.boot', name: 'spring-boot-starter-actuator'){
+        exclude(module:"micrometer-core")
```
Comment (amitchidrewar1301) [inline api/build.gradle:72]: We don't need exclusion as we have explicitly added a resolution strategy

Comment (Jennings-HMCTS) [reply api/build.gradle]: have removed the exclusion

File: api/src/main/resources/application-local.properties [hunk 3]
```diff
@@ -1,2 +1,30 @@
 auth.idam.client.baseUrl=http://localhost:8001
 enable.fee.loader=false
+
```
Comment (amitchidrewar1301) [inline api/src/main/resources/application-local.properties:3]: Please remove application local properties which are specific to your case

File: charts/fees-register-api/Chart.yaml [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
 name: fees-register-api
 home: https://github.com/hmcts/ccfr-fees-register-app
-version: 0.3.0
+version: 0.3.2
 maintainers:
   - name: HMCTS Fees & Payments Dev Team
     email: ccpay@hmcts.net
+dependencies:
+  - name: java
+    version: ~3.0.0-beta
```
Comment (timja) [inline charts/fees-register-api/Chart.yaml:12]: don't use a beta version if you don't have to this is well out of date:
https://github.com/hmcts/chart-java/releases

File: charts/fees-register-api/values.yaml [hunk 5]
```diff
@@ -1,6 +1,7 @@
 java:
   image: 'hmctspublic.azurecr.io/fees-register/api:latest'
   applicationPort: 8080
+  readinessPath: '/health/readiness'
```
Comment (timja) [inline charts/fees-register-api/values.yaml:4]: you shouldn't need to override this, the new chart has this as default

File: charts/fees-register-api/values.yaml [hunk 6]
```diff
@@ -18,6 +19,8 @@ java:
     ISSUER_URI: https://idam-web-public.{{ .Values.global.environment }}.platform.hmcts.net/o
     OIDC_ISSUER: https://forgerock-am.service.core-compute-idam-{{ .Values.global.environment }}.internal:8443/openam/oauth2/hmcts
     OIDC_AUDIENCE_LIST: fees_admin_frontend,paybubble
+    SERVER_SHUTDOWN_STRATEGY: graceful
```
Comment (timja) [inline charts/fees-register-api/values.yaml:22]: you've also set this in your application configuration, consider whether duplicating this config is the right thing to do

File: dependency-check-suppressions.xml [hunk 7]
```diff
@@ -60,5 +60,15 @@
         <notes>HTTP Client 4.5.11 suppression, root cause and fix needs to be carried out separately</notes>
         <cve>CVE-2020-13956</cve>
     </suppress>
-    
+
+    <suppress>
+        <notes>This is because of the transitive dependency fasterxml:jackson for which mitigation is not yet available</notes>
+        <cve>CVE-2020-25649</cve>
```
Comment (timja) [inline dependency-check-suppressions.xml:66]: pretty sure there is mitigations, everyone else has upgraded it

Also never add an exclusion without an expiry date

**Review by timja — CHANGES_REQUESTED** [review]
dep check and chart version are blocking comments

File: charts/fees-register-api/values.yaml (hunk 6 above)
Comment (amitchidrewar1301) [reply charts/fees-register-api/values.yaml:21]: Even if right now values for application and values.yaml are same but going forward values.yaml will help us to changes the strategy.

File: charts/fees-register-api/Chart.yaml (hunk 4 above)
Comment (amitchidrewar1301) [reply charts/fees-register-api/Chart.yaml]: The version has been upgraded now

File: charts/fees-register-api/values.yaml (hunk 5 above)
Comment (amitchidrewar1301) [reply charts/fees-register-api/values.yaml]: @timja Necessary changes have been done

File: dependency-check-suppressions.xml (hunk 7 above)
Comment (amitchidrewar1301) [reply dependency-check-suppressions.xml]: @timja As per the recommendations tomcat and Jackson databind jars are updated. Suppressions are also removed.

**Review by timja — APPROVED** [review]

**Review by YasharRahvar — APPROVED** [review]

**Comment by timja** [comment]
(Oops thought this was my PR didn’t mean to merge this one, I’ve double checked all code and all seems very safe)

**Pull request merged**